core/store/src/genesis/state_applier.rs

```diff
@@ -269,40 +270,54 @@ impl GenesisStateApplier {
-                        )
+            match receipt.receipt {
+                ReceiptEnum::Action(ref action_receipt) => {
+                    let mut pending_data_count: u32 = 0;
+                    for data_id in &action_receipt.input_data_ids {
+                        storage.modify(|state_update| {
+                            if !has_received_data(state_update, account_id, *data_id)
+                                .expect("Genesis storage error")
+                            {
+                                pending_data_count += 1;
+                                set(
+                                    state_update,
+                                    TrieKey::PostponedReceiptId {
+                                        receiver_id: account_id.clone(),
+                                        data_id: *data_id,
+                                    },
+                                    &receipt.receipt_id,
+                                )
+                            }
+                        });
                     }
-                });
-            }
-            if pending_data_count == 0 {
-                panic!("Postponed receipt should have pending data")
-            } else {
-                storage.modify(|state_update| {
-                    set(
-                        state_update,
-                        TrieKey::PendingDataCount {
-                            receiver_id: account_id.clone(),
-                            receipt_id: receipt.receipt_id,
-                        },
-                        &pending_data_count,
-                    );
-                    set_postponed_receipt(state_update, &receipt);
-                });
+                    if pending_data_count == 0 {
+                        panic!("Postponed receipt should have pending data")
+                    } else {
+                        storage.modify(|state_update| {
+                            set(
+                                state_update,
+                                TrieKey::PendingDataCount {
+                                    receiver_id: account_id.clone(),
+                                    receipt_id: receipt.receipt_id,
+                                },
+                                &pending_data_count,
+                            );
+                            set_postponed_receipt(state_update, &receipt);
+                        });
+                    }
+                }
+                ReceiptEnum::PromiseYield(ref _action_receipt) => {
+                    // TODO: check there there is exactly one input data dependency and that it is
+                    // unsatisfied
+                    storage.modify(|state_update| {
+                        set_yielded_promise(state_update, &receipt);
+                    });
+                }
+                ReceiptEnum::Data(_) | ReceiptEnum::PromiseResume(_) => {
+                    unreachable!("Expected action receipt")
```

Comment:
Q: should we perhaps use `log_assert!` here instead? What are the possible consequences of effectively no-op handling if we encounter Data/PromiseResume here?

Reply:
Not sure what the difference is between `panic!` and `log_assert!` but I went with `panic!` as it is used elsewhere in this function.